docs/namespace-scoped-setup.md

```diff
@@ -0,0 +1,206 @@
+# Namespace-Scoped Baremetal Operator (BMO) Deployment
+
+This document explains how to configure the Baremetal Operator (BMO) to operate
+within specific Kubernetes namespaces rather than across the entire cluster.
+This is crucial for multi-tenant environments or when you need to limit the
+operator's permissions and scope for security reasons.
+
+## 1. Generating Namespace-Scoped Manifests
+
+To deploy BMO with namespace-specific configurations, you need to generate
+specialized Kubernetes manifests.
+
+**How to generate:**
+
+## 1\. Configuring BMO to Watch Specific Namespaces
+
+The BMO controller itself needs to be instructed which namespaces it should
+monitor for resources.
+
+**Using the `WATCH_NAMESPACE` environment variable:**
+
+You can control the operator's watch scope by setting the `WATCH_NAMESPACE`
+environment variable in its deployment.
+
+* **Watch all namespaces (default):** If `WATCH_NAMESPACE` is unset or empty,
+  BMO will monitor resources across all namespaces in the cluster.
+* **Watch a single namespace:** Set `WATCH_NAMESPACE` to the desired
+  namespace name (e.g., `WATCH_NAMESPACE=my-namespace`).
+* **Watch multiple namespaces:** Provide a comma-separated list of namespaces
+  (e.g., `WATCH_NAMESPACE=foo,bar,quo`).
+
+**Example: Configuring BMO to watch its own deployed namespace:**
+
+To configure the operator to automatically watch the same namespace it's
+deployed in, modify the `config/base/manager.yaml` file. Add the following
```

Comment:
The example below is watching foo,bar and quo.
```suggestion
To configure the operator to automatically watch the same namespaces foo,
bar and quo, modify the `config/base/manager.yaml` file. Add the following
```

Reply:
Done as well